utils/port_utils.py

```
@@ -0,0 +1,278 @@
+        ERROR_ALREADY_EXISTS = 183
+
+        handle = kernel32.CreateMutexW(None, True, _LOCK_NAME)
+        last_err = kernel32.GetLastError()
+
+        if handle and last_err != ERROR_ALREADY_EXISTS:
+            _lock_handle = handle
+            return True
+        # 说明已有其他实例持有互斥体
+        if handle:
+            kernel32.CloseHandle(handle)
+        return False
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)